src/clzones.cpp

```diff
@@ -11,61 +11,59 @@
 #include <iostream>
 #include <fstream>
 
-clZones::clZones()
+zone_manager::zone_manager()
 {
-    //Add new zone types here
-    //Use: if (g->checkZone("ZONE_NAME", posx, posy)) {
-    //to check for a zone
-
-    vZoneTypes.push_back(std::make_pair(_("No Auto Pickup"), "NO_AUTO_PICKUP"));
+    types.push_back( std::make_pair( _("No Auto Pickup"), "NO_AUTO_PICKUP" ) );
+    types.push_back( std::make_pair( _("No NPC Pickup"),  "NO_NPC_PICKUP" ) );
 }
 
-void clZones::clZoneData::setName()
+void zone_manager::zone_data::set_name()
 {
-    const std::string sName = string_input_popup(_("Zone name:"), 55, this->sName, "", "", 15);
+    const std::string new_name = string_input_popup( _( "Zone name:" ), 55, name, "", "", 15 );
 
-    this->sName = (sName == "") ? _("<no name>") : sName;
+    name = ( new_name == "" ) ? _( "<no name>" ) : new_name;
 }
 
-void clZones::clZoneData::setZoneType(std::vector<std::pair<std::string, std::string> >
-                                      vZoneTypes)
+void zone_manager::zone_data::set_zone_type()
 {
+    const auto &types = get_manager().get_zone_types();
     uimenu as_m;
-    as_m.text = _("Select zone type:");
+    as_m.text = _( "Select zone type:" );
 
-    for (unsigned int i = 0; i < vZoneTypes.size(); ++i) {
-        as_m.entries.push_back(uimenu_entry(i + 1, true, (char)i + 1, vZoneTypes[i].first));
+    for( size_t i = 0; i < types.size(); ++i ) {
+        as_m.entries.push_back( uimenu_entry( i + 1, true, ( char )i + 1, types[i].first ) );
     }
+
     as_m.query();
 
-    this->sZoneType = vZoneTypes[((as_m.ret >= 1) ? as_m.ret : 1) - 1].second;
+    type = types[( ( as_m.ret >= 1 ) ? as_m.ret : 1 ) - 1].second;
 }
 
-void clZones::clZoneData::setEnabled(const bool p_bEnabled)
+void zone_manager::zone_data::set_enabled( const bool _enabled )
 {
-    this->bEnabled = p_bEnabled;
+    enabled = _enabled;
 }
 
-point clZones::clZoneData::getCenterPoint()
+tripoint zone_manager::zone_data::get_center_point() const
 {
-    return point((pointStartXY.x + pointEndXY.x) / 2, (pointStartXY.y + pointEndXY.y) / 2);
+    return tripoint( ( start.x + end.x ) / 2, ( start.y + end.y ) / 2, ( start.z + end.z ) / 2 );
 }
 
-std::string clZones::getNameFromType(const std::string p_sType)
+std::string zone_manager::get_name_from_type( const std::string &type ) const
 {
-    for( auto &elem : vZoneTypes ) {
-        if( elem.second == p_sType ) {
+    for( auto &elem : types ) {
+        if( elem.second == type ) {
             return elem.first;
         }
     }
 
-    return "Unknown Type";
+    return _("Unknown Type");
```

Comment:
In my opinion, this marks a bug, it should not be translated to be obvious, same reason one does not translate the debug messages.